File: src/main/resources/python/algohub/judge_result.py
import json


class JudgeResult(object):
    def __init__(self, status_code, error_message=None, input=None,
                 output=None, expected_output=None, testcase_passed_count=0,
                 testcase_total_count=0, elapsed_time=0, consumed_memory=0):
        self.status_code = status_code
        self.error_message = error_message
        self.input = input
        self.output = output
        self.expected_output = expected_output
        self.testcase_passed_count = testcase_passed_count
        self.testcase_total_count = testcase_total_count
        self.elapsed_time = elapsed_time
        self.consumed_memory = consumed_memory

    def to_json(self):
        result = '{'
        result += '"status_code":' + str(self.status_code.value)
        if self.error_message is not None:
            result += '， "error_message":' + json.dumps(self.error_message)
        if self.input is not None:
            result += '， "input":' + json.dumps(self.input)
        if self.output is not None:
            result += '， "output":' + json.dumps(self.output)
        if self.expected_output is not None:
            result += '， "expected_output":' + json.dumps(self.expected_output)
        if self.testcase_passed_count is not None and self.testcase_passed_count != 0:
            result += '， "testcase_passed_count":' + str(self.testcase_passed_count)
        if self.testcase_total_count is not None and self.testcase_total_count != 0:
            result += '， "testcase_total_count":' + str(self.testcase_total_count)
        if self.elapsed_time is not None and self.elapsed_time != 0:
            result += '， "elapsed_time":' + str(self.elapsed_time)
        if self.consumed_memory is not None and self.consumed_memory != 0:
            result += '， "consumed_memory":' + str(self.consumed_memory)
        result += '}'
        return result

File: src/main/java/org/algohub/engine/pojo/JudgeResult.java
package org.algohub.engine.pojo;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.ArrayNode;
import org.algohub.engine.bo.StatusCode;


/**
 * Judge result.
 */
@SuppressWarnings("PMD.CommentRequired")
public final class JudgeResult {
  @JsonProperty(value = "status_code", required = true) private StatusCode statusCode;
  @JsonProperty("error_message") private String errorMessage;
  private ArrayNode input;
  private JsonNode output;
  @JsonProperty("expected_output") private JsonNode expectedOutput;
  @JsonProperty("testcase_passed_count") private int testcasePassedCount;
  @JsonProperty("testcase_total_count") private int testcaseTotalCount;
  @JsonProperty("elapsed_time") private long elapsedTime; // milliseconds
  @JsonProperty("consumed_memory") private long consumedMemory;  // bytes

  /**
   * Since some fields are immutable, need to provide a method for Jackson.
   */
  @JsonCreator public JudgeResult(@JsonProperty(value = "status_code", required = true) final StatusCode statusCode,
      @JsonProperty("error_message") final String errorMessage,
      @JsonProperty("input") final ArrayNode input, @JsonProperty("output") final JsonNode output,
      @JsonProperty("expected_output") final JsonNode expectedOutput,
      @JsonProperty("testcase_passed_count") final int testcasePassedCount,
      @JsonProperty("testcase_total_count") final int testcaseTotalCount,
      @JsonProperty("elapsed_time") final long elapsedTime,
      @JsonProperty("consumed_memory") final long consumedMemory) {
    this.statusCode = statusCode;
    this.errorMessage = errorMessage;
    this.input = input;
    this.output = output;
    this.expectedOutput = expectedOutput;
    this.testcasePassedCount = testcasePassedCount;
    this.testcaseTotalCount = testcaseTotalCount;
    this.elapsedTime = elapsedTime;
    this.consumedMemory = consumedMemory;
  }

  /**
   * Constructor.
   */
  public JudgeResult(final String compileErrorMsg) {
    this.statusCode = StatusCode.COMPILE_ERROR;
    this.errorMessage = compileErrorMsg;
  }

  // PENDING or RUNNING
  public JudgeResult(final StatusCode statusCode) {
    this.statusCode = statusCode;
  }

  public StatusCode getStatusCode() {
    return statusCode;
  }

  public String getErrorMessage() {
    return errorMessage;
  }

  public void setErrorMessage(String errorMessage) {
    this.errorMessage = errorMessage;
  }

  public ArrayNode getInput() {
    return input;
  }

  public JsonNode getOutput() {
    return output;
  }

  public JsonNode getExpectedOutput() {
    return expectedOutput;
  }

  public int getTestcasePassedCount() {
    return testcasePassedCount;
  }

  public int getTestcaseTotalCount() {
    return testcaseTotalCount;
  }

  public void setTestcaseTotalCount(int n) {
    this.testcaseTotalCount = n;
  }

  public long getElapsedTime() {
    return elapsedTime;
  }

  public long getConsumedMemory() {
    return consumedMemory;
  }

  public void setElapsedTime(long elapsedTime) {
    this.elapsedTime = elapsedTime;
  }
}
